src/blog/2024/06/using-node-red-as-an-etl-tool.md

```diff
@@ -0,0 +1,231 @@
+
+*Note: The goal of this project is to understand how to utilize Node-RED as an ETL tool. We assume that the reader has a basic knowledge of Node-RED.*
+
+## Prerequisite 
+
+Before proceeding further make sure you have installed the following nodes:
+
+1. [node-red-contrib-mongodb4](https://flows.nodered.org/node/node-red-contrib-mongodb4)
+2. [@flowfuse/node-red-dashboard](https://dashboard.flowfuse.com/)
+
+## Extracting Data
+
+1. Drag an Inject node onto the canvas.
+2. Drag an HTTP Request node onto the canvas, double-click on it, and set the URL to `https://api.slingacademy.com/v1/sample-data/files/customers.json`.
+3. Drag a Link-out node onto the canvas, I am using Link nodes to separate flow groups for different ETL phases to enhance organization, although you can directly connect nodes or maintain a single flow group.
+4. Connect the Inject node's output to the input of the HTTP Request node and the HTTP Request node's output to the input of the Link-out node.
+
+Now, if you add a Debug node after the HTTP Request node, you will see an array of objects containing customer data in the Debug panel. Each object has properties such as `first_name`, `last_name`, `email`, and others.
+
+!["Screenshot of debug panel displaying printed customer data"](./images/etl-with-node-red-chart-customer-data-on-debug-panel.png "Screenshot of debug panel displaying printed customer data"){data-zoomable}
+
+In next section, we'll perform a transformation to replace the existing properties `first_name` and `last_name` with a single property `full_name` that concatenates them.
+
+## Transforming Data
+
+### Data Cleaning
+
+1. Drag a Link-in node onto the canvas, double-click on it, and select the name of the Link-out node added in the above section.
+2. Drag a Split node onto the canvas.
+3. Drag a Change node onto the canvas and Set `msg.payload.full_name` to `payload.first_name & " " & payload.last_name` as the JSONata expression, this JSONata expression will concat the firstname and lastname properties, then delete unnecessary properties like `first_name` and `last_name`.
+4. Drag a Join node onto the canvas.
+5. Connect the output of the Link-in node to the input of the Split node, then the Split node output to the input of the Join node.
+
+Up to this point, we've established a flow that creates a new property in customer data objects called 'full_name' and subsequently removes any unnecessary properties. In summary, we've cleaned the data according to our needs.
+
+### Customer Data Deduplication
+
+1. Drag a Change node onto the canvas, and set `msg.data` to `msg.payload` and `msg.payload` to `[]`.
+2. Drag a MongoDB4 node onto the canvas, and configure it with your correct details. If you haven't used MongoDB with Node-RED, please refer to the [Using MongoDB with Node-RED](/blog/2024/04/using-mongodb-with-node-red/). Enter `find` into the Operation field.
```

Comment:
I don't think MongoDB is used very often in an ETL context, precisely as it's structureless. Do we have a better example? An event based sensor perhaps?